### ISSUE TITLE
Sandbox API is deprecated

### ISSUE THREAD

**Issue by terribleplan** [ruimarinho/authy-client#34]
According to conversations I've had with twilio/authy the sandbox API is deprecated.

Due to this I implemented a mocked authy server for testing purposes, but I was unable to use this library without nasty workarounds due to the (in my opinion) overly strict validation of the hosts being used.

At the very least you should note that the sandbox is deprecated, or prevent people from using it outright, and I would greatly appreciate it if you allowed for other endpoints than the official one.

I could whip up a PR for this if you tell me which route is preferable, but it seems like it would be a simple fix either way.

**Comment by ruimarinho** [comment]
Tough choice. The validation avoids typos and other similar errors when configuring the client. It won't be that much different than connecting to a non-existing server, but it's one less request with a specific error and a non-cryptic message.

I guess we could be looking at two PRs - one that adds the note on the README and another one removing the choice assertion. I'll merge the first when you're ready and give the second one another thought.

**Comment by ruimarinho** [comment]
@josh-authy can you provide some insight here?

**Comment by ruimarinho** [comment]
No additional feedback from Authy. @terribleplan closing until we have official word from Authy on this.

**Comment by josh-authy** [comment]
Sorry, just saw this.   There is no longer a sandbox.  

Our suggestion is to create multiple Authy accounts (YOURAPP-DEV & YOURAPP) and to use the associated API keys in each respective location.  You can also leverage sub-accounts via the Twilio console if you want to really segregate users and apps.   @ruimarinho @terribleplan

**Comment by ruimarinho** [comment]
@josh-authy the current sandbox API continues to reply successfully just as it used to though. Is there any active deprecation plan?

**Comment by josh-authy** [comment]
We have some customers who use it as part of an old build cycle so no
active deprecation plans AFAIK TO

 We discourage it's use as it provides no real value IMO.

**Comment by ruimarinho** [comment]
Ok thanks, added a PR to lift that restriction up.

**Comment by challamzinniagroup** [comment]
Dirtying up your production account with test data.
Spinning an entire second account to use as a pseduo-sandbox to keep your production data clean.

These are the options we're being offered, and you say "sandbox offers no value"...? wow...

**Comment by josh-authy** [comment]
Have you actually gone through this process?   When you do, you'll notice it takes only a few seconds to create an additional Verify or Authy app.  We'd suggest naming them appropriately.  YOUR_APP, YOUR_APP QA, YOUR_APP DEV.   You'll have a unique API key for each allowing you to bin data separately.  

No, you don't spin up a second account.   No, you do not dirty your production account with test data.

You can also create a sub-account or even a Project if you'd like to segment your data and traffic even further.

**Comment by challamzinniagroup** [comment]
Yes, I have gone through the process. I don't make it a habit to comment publicly on topics I know nothing about. Twilio SMS/Voice APIs all provide LIVE/TEST creds. Authy seems to consider their own "no benefit from" opinion of sandbox - not their customers. It's not that big of a deal to run separate Authy apps - it just _feel wrong_. Like something a junior dev does before they realize 99% of all APIs out there actually offer sandbox functionality.

**Comment by josh-authy** [comment]
How do you think a sandbox would work with Authy?

We have an example testing framework you can use for CI found here:
https://github.com/AuthySE/authy-testing-sample

**Comment by challamzinniagroup** [comment]
In the scope of work I've just come off, I used:
- Twilio API for SMS send only
- Authy API to create User records
- Authy API to fetch User status
- Authy API to gen User QR codes to link Authy user record with Authenticator app
- Authy API to validate TOTP

While I can certainly spin dual Authy apps within the Twilio account to accomplish dev/testing of the Authy API calls mentioned, this is far less than ideal for a few reasons:
- It is very common to trickle down and sanitize Dbs from Prod->Staging->Dev->Local. In a dual Authy app scenario; this breaks the Authy Ids between any environments that are configured to hit different Authy apps.
- The lower the environment, the higher quantity of "ancillary user accounts" that get created. By this I mean Prod will have only real accounts. Staging will have real plus some dev/testing accounts, and dev/local environments will have a high number of test, dummy, and garbage accounts. I hope I don't need to explain how or why it is bad to have all these accounts cluttering up the primary, prod Authy app User records. Which would happen with a single Authy app.

What I ended up having to do was a dual Authy app scenario; and add code, which lives on in the code base, to check every user AuthyId and against that environment's Authy app and re-fetch if it doesn't match.

If I could have simply plugged in a sandbox Url, or token, then all Authy user records could share an AuthyId between environments and yet remain segregated between prod and sandbox. It's rather unfortunate that I have to explain this; this is production-system 101 imo.
